hw1-app/src/app/courses/course-item/course-item.component.ts

```diff
@@ -1,25 +1,74 @@
-import { Component, OnInit, ChangeDetectionStrategy } from '@angular/core';
-import { ICourse, Course } from '../models/course-item.model';
+import { Component,
+  OnInit,
+  ChangeDetectionStrategy,
+  Input,
+  EventEmitter,
+  Output,
+  OnChanges,
+  DoCheck,
+  AfterContentInit,
+  AfterContentChecked,
+  AfterViewInit,
+  AfterViewChecked,
+  OnDestroy } from '@angular/core';
+import { ICourse } from '../models/course-item.model';
 
 @Component({
   selector: 'course-item',
   templateUrl: './course-item.component.html',
   styleUrls: ['./course-item.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class CourseItemComponent implements OnInit {
-  public item: ICourse = new Course({
-    id: 'id',
-    title: 'title',
-    creationDate: new Date(),
-    duration: 33,
-    description: 'description'
-  });
+export class CourseItemComponent implements OnInit,
+    OnChanges,
+    DoCheck,
+    AfterContentInit,
+    AfterContentChecked,
+    AfterViewInit,
+    AfterViewChecked,
+    OnDestroy {
+  @Input() item: any = {};
```

Comment:
Improve you type set ICourse. Why do you need {}, maybe nulll instead

Reply:
We cannot assign to Course Item null, or even {}, if we set ICourse type, I will create an empty instance of Course